test/Microsoft.Extensions.Http.Test/DependencyInjection/HttpClientFactoryServiceCollectionExtensionsTest.cs

```diff
@@ -518,6 +538,57 @@ public void AddHttpClient_WithTypedClientAndImplementation_AndServiceDelegate_Co
             Assert.Equal("http://example.com/", client.HttpClient.BaseAddress.AbsoluteUri);
         }
 
+        [Fact]
+        public void AddHttpMessageHandler_WithName_NewHandlerIsSurroundedByLogging_ForHttpMessageHandler()
+        {
+            var serviceCollection = new ServiceCollection();
+
+            HttpMessageHandlerBuilder builder = null;
+
+            // Act1
+            serviceCollection.AddHttpClient("example.com").ConfigureHttpMessageHandlerBuilder(b =>
+            {
+                builder = b;
+
+                b.AdditionalHandlers.Add(Mock.Of<DelegatingHandler>());
+            });
+
+            var services = serviceCollection.BuildServiceProvider();
+
+            var factory = services.GetRequiredService<IHttpMessageHandlerFactory>();
+
+            // Act2
+            var handler = factory.CreateHandler("example.com");
+
+            // Assert
+            Assert.NotNull(handler);
+            Assert.IsNotType<LoggingScopeHttpMessageHandler>(handler);
+
+            Assert.Collection(
+                builder.AdditionalHandlers,
+                h => Assert.IsType<LoggingScopeHttpMessageHandler>(h),
+                h => Assert.NotNull(h),
+                h => Assert.IsType<LoggingHttpMessageHandler>(h));
+        }
+
+        [Fact] // Verifies IHttpClientFactory and IHttpMessageHandlerFactory are backed by the same internal handlers
+        public void AddHttpClient_ProvidesSameImplementationForClientsAndHandlers()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+
+            // Act
+            serviceCollection.AddHttpClient();
+
+            var services = serviceCollection.BuildServiceProvider();
+
+            var clientFactory = services.GetRequiredService<IHttpClientFactory>();
+            var handlerFactory = services.GetRequiredService<IHttpMessageHandlerFactory>();
+
+            // Assert
+            Assert.Same(clientFactory, handlerFactory);
+        }
+
```

Comment:
I'd also like to see a unit test somewhere that verifies that disposing a handler returned from the factory doesn't dispose the handler chain. I don't doubt that the code is correct, but I want a test since that's a very important detail and we need to preserve it.

Reply:
I've added another test case to `DefaultHttpClientFactoryTest` where there was already one to test the disposable behaviour of the `HttpClient`.